aws/resource_aws_wafregional_web_acl_association_test.go

```diff
@@ -0,0 +1,161 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/terraform"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/waf"
+	"github.com/aws/aws-sdk-go/service/wafregional"
+)
+
+func TestAccAWSWafRegionalWebAclAssociation_basic(t *testing.T) {
+	var webAcl waf.WebACL
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckWafRegionalWebAclAssociationDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckWafRegionalWebAclAssociationConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckWafRegionalWebAclAssociationExists("aws_wafregional_web_acl_association.foo", &webAcl),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckWafRegionalWebAclAssociationDestroy(s *terraform.State) error {
+	return testAccCheckWafRegionalWebAclAssociationDestroyWithProvider(s, testAccProvider)
+}
+
+func testAccCheckWafRegionalWebAclAssociationDestroyWithProvider(s *terraform.State, provider *schema.Provider) error {
+	conn := provider.Meta().(*AWSClient).wafregionalconn
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_wafregional_web_acl_association" {
+			continue
+		}
+
+		web_acl_id, resource_arn := resourceAwsWafRegionalWebAclAssociationParseId(rs.Primary.ID)
```

Comment:
Nitpick: it is a convention in Go generally to follow `camelCase` instead of `under_score` for variables.